ITSNP-3: Design Card components

bijudhungel11 opened the pull request:
- Designed posts trending component card
- created resources folder under src which contains images
- created constants folder for importing all images from resources folder
- made mixin for font and display item in flex
- shame: design is not responsive

bijudhungel11 committed:
Design Card components
- Designed posts trending component card
- created resources folder under src which contains images
- created constants folder for importing all images from resources folder
- made mixin for font and display item in flex
- shame: design is not responsive

SurrealSerenade commented:
@bijudhungel11 Please link the issue related to this pull request (PR). Also, name the PR as ITSNP-03: Design Card Components. Thank you.

SurrealSerenade commented on public/index.html:
```diff
@@ -24,6 +24,11 @@
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
+    <link rel="preconnect" href="https://fonts.gstatic.com" />
+    <link
```
It is not preferred to add fonts to index.html. I think there is a _fonts.scss file where you can add fonts to.

bijudhungel11 replied:
Completed

SurrealSerenade commented on src/components/PostsTrendingCard/PostsTrendingCard.js:
```diff
@@ -0,0 +1,28 @@
+import { allImage } from "../../constants";
```
The naming convention is to use capital letters.

SurrealSerenade commented on src/components/PostsTrendingCard/PostsTrendingCard.js:
```diff
@@ -0,0 +1,28 @@
+    <div className="postsTrendingCard">
+      <div>
+        <img
+          src={allImage.momo}
```
You can check out this [PR](https://github.com/itsnporg/blog-itsnp-frontend/pull/10/files#diff-6b6fe6d5d64274a3f623c5e7b5240b68e93fda3cb6b8353e892c49af03d0ea68) to create components that load content based on props provided to it.

bijudhungel11 replied:
Done

SurrealSerenade commented on src/components/PostsTrendingCard/PostsTrendingCard.scss:
```diff
@@ -0,0 +1,29 @@
+  @include flex-item-center(row);
+  width: 594px;
+
+  &__img {
```
There is already a mixin that creates an element (BEM), you can use that to create the element.

SurrealSerenade commented on src/components/PostsTrendingCard/PostsTrendingCard.scss:
```diff
@@ -0,0 +1,29 @@
+
+  &__img {
+    height: 141px;
+}
```
Fix indentation.

bijudhungel11 replied:
Used prettier

SurrealSerenade commented on src/components/PostsTrendingCard/PostsTrendingCard.scss:
```diff
@@ -0,0 +1,29 @@
+    height: 141px;
+}
+
+  &__content {
```
There is already a mixin that creates an element (BEM), you can use that to create the element.
The code for creating element must look like: `@include e('content') { ... }`

bijudhungel11 replied:
Done

SurrealSerenade commented on src/constants/Image.js:
```diff
@@ -0,0 +1,22 @@
+
+/* icons */
+
+import bookmark from "../resources/images/icons/bookmark.png";
```
Icons can be retrieved from Figma as SVG and components can be created for them. You can check out this [PR](https://github.com/itsnporg/blog-itsnp-frontend/pull/10/files#diff-93aa9ea357a87a6cd8511b0e91cbe1685f15d7ebc730e61273ae18d484ae8cdd) for more information.

RudeSoul commented:
I think this is the first thing you need to fix,

![image](https://user-images.githubusercontent.com/30189506/121345237-54a52400-c944-11eb-803c-7864ca045bcf.png)

RudeSoul commented on public/index.html:
```diff
@@ -24,6 +24,11 @@
+    <link
+      href="https://fonts.googleapis.com/css2?family=Poppins:wght@300&display=swap"
+      rel="stylesheet"
+    />
```
` <link
      href="https://fonts.googleapis.com/css2?family=Poppins:wght@300&display=swap"
      rel="stylesheet"
    />`

we have _font.scss file for these things

SurrealSerenade commented on src/styles/utils/_mixins.scss:
```diff
@@ -31,3 +31,27 @@
+  flex-direction:$direction;
+}
+
+@mixin font-heading() {
```
Mixins should be reusable and general. You can use parameters to mixins to make them reusable. The way it is right now is static and may conflict with other's requirements.

bijudhungel11 replied:
Deleted those mixin which was not reusable.

RudeSoul commented on src/components/PostsTrendingCard/PostsTrendingCard.js:
```diff
@@ -0,0 +1,28 @@
+      <div className="postsTrendingCard__content">
+        <h1>How momo helps to think like bill gates</h1>
+        <h3>
+          I used to have 5 plates momo per day, Which helps me to build
```
this data are dynamic, need to receive from props

SurrealSerenade reviewed (changes requested):
Please, have a look at the comments. Thank you.

RudeSoul commented on src/pages/Home/Home.js:
```diff
@@ -1,11 +1,12 @@
-import React from 'react';
+import PostsTrendingCard from "components/PostsTrendingCard/PostsTrendingCard";
+import React from "react";
```
this must be on 1st line

RudeSoul reviewed (changes requested):
Nice pr, just some issues, Hope you will resolve asap

bijudhungel11 committed:
Design Card components (#11)
- removed font family link tag from index.html
- provided data through  props in posts Trending card component
- added imported react from library to the top of component

bijudhungel11 committed:
Design Card components #11
- used naming convnetion to capitalize the allImage to AllImage
- created icons folder to keep the all svg icons as component
- used BEM mixins in post trednindg card componenet scss file
- tried to created reusable mixin for fonts

bijudhungel11 committed:
Designed Card Component post trending card component  #3
- changed font-weight-family mixin to font
- try to make reusable component
- removed flex-item-center

Commit:
Design Component
- have completed BlogPostCard
- have completed SmalCardBlogPost Card and done it's style as welll
- have comleted MediumCardBlogPost Card
- imported all images from resources folder
- shame: Overall Design is not completed and any other mixins has not been made
- shame: haven't work deeply need to fix lot's of stuff
- shame: haven't exported image directly from Image.js as we are guided to export it directly from Image.js

bijudhungel11 committed:
Refactor  Blogpostcard component
- changed image imoprt to svg component import
- tried to provide the props for all dyanamic data
- changed some scss to make pixel perfect
- shame: still some scss is not refactored

sajinshrestha2015 committed:
Design Card Component
- resolve Images

bijudhungel11 committed:
Designed card componenet #3 Refactor
- Removed constants folder to import all images
- directly importing from the resources folder to the component

ace4port commented on src/pages/Home/Home.js:
```diff
@@ -1,11 +1,51 @@
+    <div className="homePage">
+      <PostsTrendingCard
+        image={MOMO}
+        caption="momo"
```
Why is the image caption name `caption `here and elsewhere `imagetitle`?
Please maintain consistency in naming

bijudhungel11 replied:
changed the caption  to imageTitle in all component

ace4port commented on src/components/BlogPostCard/index.js:
```diff
@@ -0,0 +1,43 @@
+        </div>
+        <div className="blogPostCard__circle">
+          <span>{userName}</span>
+          <img src={user} alt="" className="blogPostCard__circle--user" />
```
This is not needed. You could use `Avatar` component from profile components. Just pass in size props as xs

Also _naming_ - user is not appropriate. The image used here will be of author of post not the user who logged in.

bijudhungel11 replied:
Avatar component is added.

ace4port commented on src/components/Circles/LargePink.js:
```diff
@@ -0,0 +1,17 @@
+import React from "react";
```
This code is difficult to understand. Even you, the **writer** will have difficulty understanding it if you open this code 2 weeks or a month from now.  These are design elements , not used anywhere else in design- so making them components is not required. 
All elements in this - `Largepink` `purple` `smallpink` They are used once- for background of "Join Madhyam" This all should be part of that component or in the same folder at the very least

bijudhungel11 replied:
Moved the circles folder which was on the components folder to joinMadhyam components.

bijudhungel11 committed:
Resolved issue in desinged card compenent #3
- changed caption naming convention for alternative attribut in image tag with imageTitle props
- tried to use Avatar component which was call from  profileComp branch made by ace4port
- moved the circles folder to joinMadhyam component folder where it belongs

SurrealSerenade commented on src/components/BlogPostCard/index.js:
```diff
@@ -0,0 +1,45 @@
+  imageTitle,
+  authorImg,
+  date = "May 17",
+  authorName = "XettriAl.",
```
Does not make much sense to have default values for props `date` and `authorName` as "May 17" and "XettriAl.".

SurrealSerenade commented on src/components/JoinMadhyamCard/index.js:
```diff
@@ -0,0 +1,41 @@
+import SmallPink from "./Circles/SmallPink";
+import Purple from "./Circles/Purple";
+const JoinMadhyamCard = ({
+  topic = "Join Mnadhyam!",
```
Please correct this spelling.

SurrealSerenade reviewed (commented):
Exceptional Work, Keep it up! Do fix the minor comments though.

sajinshrestha2015 committed:
Minor problem has been fixed
